index.html

```diff
@@ -303,7 +303,7 @@ <h3>Normative Specifications</h3>
                     <dt id="hash" class="spec">RDF Dataset Hash (RDH)</dt>
                     <dd>
                         <p>
-                            This specification details the processing steps of a hash function for an arbitrary RDF Dataset. These step include (1) the generation of a <a href="./explainer.html#canonical_form">canonical form</a> using the algorithm specified in the “RDF Dataset Canonicalization” deliverable, (2) sorting the <a href="https://www.w3.org/TR/n-quads/">N-Quads</a> serialization of the canonical form generated in the previous step, and (3) application of a cryptographic hash function on the sorted <a href="https://www.w3.org/TR/n-quads/">N-Quads</a> representation. 
+                            This specification details the processing steps of a hash function for an arbitrary RDF Dataset. These step include (1) the generation of a <a href="./explainer.html#canonical_form">canonical form</a> for the RDF Dataset using the algorithm specified in the “RDF Dataset Canonicalization” deliverable, (2) application of a hash function over the canonical form of the RDF Dataset. 
```

Comment:
Well, the change here seems to modify the definition of "a hash function" to include serialization of the canonicalized dataset, whereas I believe we were going for "hash function" to be something where we could slot in any one-way cryptographic function. Can we do this instead as a middleground:

```suggestion
                            This specification details the processing steps of a hash function for an arbitrary RDF Dataset. These step include (1) the generation of a <a href="./explainer.html#canonical_form">canonical form</a> for the RDF Dataset using the algorithm specified in the “RDF Dataset Canonicalization” deliverable, (2) application of a hash function over a serialization (e.g., sorted <a href="https://www.w3.org/TR/n-quads/">N-Quads</a>) of the canonical form of the RDF Dataset.
```

This middleground would only mention sorted N-Quads as an example serialization, but makes it clear that the hash function takes a serialization so that any already standardized hash function can be used here.

Reply:
(I think I put this PR in the wrong place and should have added it as part of PR 92, but we can maybe agree here for the moment and then add the consensus change to PR92?)

@dlongley, I think the change you propose might exclude what I think should the best solution for RDH, which would be to hash a serialisation of each triple/quad separately (rather than the entire dataset), and then combine the hashes in an appropriate (comutative/associative) way, e.g., by XORing the hashes of each triple/quad as you do a pass once over the file. This would avoid two practical issues, particularly when computing a hash over a very large RDF Dataset: (1) having to sort the dataset; (2) having to feed the serialisation of the entire RDF dataset into a hash function. 

But maybe I am reading too much into it? I guess the process could receive a serialisation of the dataset, and then just read the dataset statement-by-statement, computing the hash that way?

Reply:
I'm happy for us to find a way to allow us to explore any of these options in the WG. I think if we say that we're going to hash some serialization -- that that's good enough here. I don't think we're being too prescriptive ... and we can point back to this discussion in the WG later to indicate all options are in scope. What I wanted to avoid was people thinking we were going to invent new hashing functions vs. use existing ones.

Btw, using the mechanism you describe would involve computing 1 hash per quad -- which, for large datasets (e.g., millions of quads?) you're now talking about computing millions of hashes, which has its own disadvantages. But, again, this would be something to really explore in the WG.

I think any further discussion should happen on #92, where other changes to the text have already been proposed.